src/core/network/qgsnetworkdiskcache.cpp

```diff
@@ -111,3 +114,75 @@ void QgsNetworkDiskCache::clear()
   const QMutexLocker lock( &sDiskCacheMutex );
   return sDiskCache.clear();
 }
+
+qint64 QgsNetworkDiskCache::smartCacheSize( const QString &cacheDir )
+{
+  static qint64 cacheSize = 0;
+  static std::once_flag initialized;
+  std::call_once( initialized, [ = ]
+  {
+    std::function<qint64( const QString & )> dirSize;
+    dirSize = [&dirSize]( const QString & dirPath ) -> qint64
+    {
+      qint64 size = 0;
+      QDir dir( dirPath );
+
+      const QStringList filePaths = dir.entryList( QDir::Files | QDir::System | QDir::Hidden );
+      for ( const QString &filePath : filePaths )
+      {
+        QFileInfo fi( dir, filePath );
+        size += fi.size();
+      }
+
+      const QStringList childDirPaths = dir.entryList( QDir::Dirs | QDir::NoDotAndDotDot | QDir::System | QDir::Hidden );
+      for ( const QString &childDirPath : childDirPaths )
+      {
+        size += dirSize( dirPath + QDir::separator() + childDirPath );
+      }
+
+      return size;
+    };
+
+    qint64 bytesFree;
+    QStorageInfo storageInfo( cacheDir );
+    bytesFree = storageInfo.bytesFree() + dirSize( cacheDir );
+
+    // NOLINTBEGIN(bugprone-narrowing-conversions)
+    // Logic taken from Firefox's smart cache size handling
+    qint64 available10MB = bytesFree / 1024 / ( 1024LL * 10 );
+    if ( available10MB > 2500 )
+    {
+      // Cap the cache size to 1GB
+      cacheSize = 100;
+    }
+    else
+    {
+      if ( available10MB > 700 )
+      {
+        // Add 2.5% of the free space above 7GB
+        cacheSize += ( available10MB - 700 ) * 0.025;
+        available10MB = 700;
+      }
+      if ( available10MB > 50 )
+      {
+        // Add 7.5% of free sapce between 500MB to 7GB
+        cacheSize += ( available10MB - 50 ) * 0.075;
+        available10MB = 50;
+      }
+
+#if defined( Q_OS_ANDROID )
+      // On Android, smaller/older devices may have very little storage
+
+      // Add 16% of free space up to 500 MB
+      cacheSize += std::max( 2LL, static_cast<qint64>( available10MB * 0.16 ) );
+#else
+      // Add 30% of free space up to 500 MB
+      cacheSize += std::max( 5LL, static_cast<qint64>( available10MB * 0.30 ) );
+#endif
+    }
```

Comment:
could you create instead a local cacheSize10MB variable in this block for more clarity ?